tdi_python/tdiDefs.py

```diff
@@ -0,0 +1,131 @@
+from enum import Enum, auto
+
+'''
+    These maps are based on the enums defined in tdi_defs.h If the
+    enums there are changed, these maps must also be changed.
+'''
+
+# tdi_target_e
+class TargetEnum(Enum):
+  TDI_TARGET_CORE = 0
+  TDI_TARGET_ARCH = 0x08
+  TDI_TARGET_DEVICE = 0x80
+
+# tdi_target_core_enum_e
+class TargetType:
+    class TargetEnumCore(Enum):
+        TDI_TARGET_DEV_ID = TargetEnum.TDI_TARGET_CORE.value
+
+    target_type_dict = {
+        TargetEnumCore.TDI_TARGET_DEV_ID.value: "dev_id"
+    }
+
+    @classmethod
+    def target_type_map(cls, target_type_enum=None, target_type_str=None):
+        if target_type_enum is not None:
+            return cls.target_type_dict[target_type_enum]
+        if target_type_str is not None:
+            return cls.target_type_rev_dict[target_type_str]
+
+
+# tdi_attributes_type_e
+class AttributesEnum(Enum):
+    TDI_ATTRIBUTES_TYPE_CORE = 0
+    TDI_ATTRIBUTES_TYPE_ARCH = 0x08
+    TDI_ATTRIBUTES_TYPE_DEVICE = 0x80
+
+class AttributesType:
+    # Attributes defined in target
+    attributes_dict = {}
+
+# tdi_operations_type_e
+class OperationsEnum(Enum):
+  TDI_OPERATIONS_TYPE_CORE = 0
+  TDI_OPERATIONS_TYPE_ARCH = 0x08
+  TDI_OPERATIONS_TYPE_DEVICE = 0x80
+
+class OperationsType:
+    # Operations defined in target'''
+    operations_dict = {}
+
+# tdi_match_type_e
+class KeyMatchEnum(Enum):
+    TDI_MATCH_TYPE_CORE = 0
+    TDI_MATCH_TYPE_ARCH = 0x08
+    TDI_MATCH_TYPE_DEVICE = 0x80
+
+# tdi_match_type_core_e
+class KeyMatchType:
+    class KeyMatchEnumCore(Enum):
+        TDI_MATCH_TYPE_EXACT = KeyMatchEnum.TDI_MATCH_TYPE_CORE.value
+        TDI_MATCH_TYPE_TERNARY = auto()
+        TDI_MATCH_TYPE_LPM = auto()
+        TDI_MATCH_TYPE_RANGE = auto()
+        TDI_MATCH_TYPE_OPTIONAL = auto()
```

Comment:
TDI_PSA_MATCH_TYPE_OPTIONAL for PSA arch type

Reply:
As discussed, we are having only two levels of hierarchy. target/arch independent code in TDI. Dependent code in target repos.